dkc/core/models/file.py

```diff
@@ -84,12 +84,8 @@ def has_permission(self, user: User, permission: Permission) -> bool:
 
 @receiver(models.signals.pre_save, sender=File)
 def _file_pre_save(sender: Type[File], instance: File, **kwargs):
-    # TODO if we allow changing a file's blob, we also need to update the size
-
+    # TODO if we allow changing a file's blob & size, we'll need more logic here
     if not instance.pk:
-        # TODO: Test how this works in minio
-        instance.size = instance.blob.size
```

Comment:
We are currently trusting users that the blob's actual size is the same as the declared size.

In the future, we should be able to depend on S3FF to authoritatively guarantee the exact size of the uploaded object (currently, this logic isn't complete in S3FF, but it can be finished) after the upload completes. However, for full functionality here, we still need to link a particular S3FF upload to this particular field *in advance* (which is https://github.com/girder/django-s3-file-field/issues/204).

As a temporary alternative to at least maintain the accuracy of `instance.size`, we could continue to re-set or re-verify the actual blob size `instance.blob.size` (which calls out to S3) vis-a-vis the database size `instance.size` any time the blob is changed.

Reply:
#139 also encompasses this issue.